blob/blob.go

```diff
@@ -611,6 +612,28 @@ func (i *ListIterator) Next(ctx context.Context) (*ListObject, error) {
 	return i.Next(ctx)
 }
 
+// All iterates over the iterator, returning a *ListObject and a download function for each entry.
+func (i *ListIterator) All(ctx context.Context, err *error) iter.Seq2[*ListObject, func(io.Writer, *ReaderOptions) error] {
```

Comment:
You need to say more about the err arg. 
But there is a better way: instead of the arg, return a `func() error`.
- The compiler won't let you ignore it, as it would with the arg.
- You can catch early calls (before the iterator is done).